quickwit-backward-compat/tests/split_metadata_non_regression.rs

```diff
@@ -0,0 +1,102 @@
+// Copyright (C) 2021 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::collections::HashMap;
+use std::fs;
+use std::path::Path;
+use std::str::FromStr;
+
+use anyhow::Context;
+use once_cell::sync::OnceCell;
+use quickwit_metastore::{SplitMetadata, SplitMetadataAndFooterOffsets, SplitState};
+
+fn check_version(
+    version: usize,
```

Comment:
I don't get this, why do we store and compare per version number, the idea is to be able to have old versions be compatible with new versions?

Also relevant may be to be able read newer version, depending on how we roll out updates
(may be hard to test ;)

Reply:
Right now it works but that is dumb actually.

I think we will have one expected splitmetadata to fill per version. 
We could save it in another file or something like that.